src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -356,6 +355,12 @@ def execute(self, args):
             table_transformer=self.commands_loader.command_table[parsed_args.command].table_transformer,
             is_query_active=self.data['query_active'])
 
+    @staticmethod
+    def _extract_parameter_names(args):
+        # note: name start with more than 2 '-' will be treated as value e.g. certs in PEM format
+        return [(p.split('=', 1)[0] if p.startswith('--') else p[:2]) for p in args if
```

Comment:
I searched a while for data models where the parameter names are already isolated from the values, but have not found out anything. So, I am still keeping the raw searching logics